### PR TITLE
Add the value of a Dialog's portalClassName prop to the React Modal portal DIV class attribute

### PR DESCRIPTION
**Pull request by stevelandwehr**
#### Screens

This PR takes the value of a Dialog's `portalClassName` prop:

image

and adds it to the React Modal portal's `div` class attribute:

image

This will allow a specific Dialog's modal to be targeted for CSS or JavaScript purposes.

#### Breaking Changes

No breaking changes. If the `className` prop isn't present then it's not added to the list of CSS class names.

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/BOP-968

#### How to test this

[INSERT TESTING DETAILS]

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **URGENCY** Please select a release milestone
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx
Comment (coleerikson) [inline playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx:58]: Thanks for the PR @stevelandwehr. At first glance I think there is probably a better way to write this with `classnames`. Take a look at: https://github.com/JedWatson/classnames#usage

`classnames('pb_dialog', buildCss('pb_dialog', size), { buildCss('pb_dialog', className): className })` potentially, but even that is a bit rough. @jasperfurniss Thoughts

**Comment by kevinschlabach** [comment]
@coleerikson @thestephenmarshall @stevelandwehr - can we please discuss what we need to for consensus and keep this moving along? It's quickly becoming the blocker for our priority Runway funnel of work and I know it will take time to get this into a Playbook release and tested/merged into Nitro still.

For example: Is the "better way" required to merge this? or can we ship this as is to keep moving?

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx:58]: `className` needs to be last on this list or CSS modules will not work. Can you correct this?

**Review by jasoncypret — CHANGES_REQUESTED** [review]

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx
Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx:58]: @stevelandwehr I just messaged you in connect over this.

**Comment by jasperfurniss** [comment]
> @coleerikson @thestephenmarshall @stevelandwehr - can we please discuss what we need to for consensus and keep this moving along? It's quickly becoming the blocker for our priority Runway funnel of work and I know it will take time to get this into a Playbook release and tested/merged into Nitro still.
> 
> For example: Is the "better way" required to merge this? or can we ship this as is to keep moving?

@kevinschlabach there's some issues we need to resolve before getting this in. I can sync with Steve and unblock him. This is a complex kit and I need to explain some of the underlying architecture to ensure that we do this correctly.

**Comment by stevelandwehr** [comment]
I paired with @jasperfurniss on this and he suggested passing the `portalClassName` prop to the `Modal` and this works like a charm. Thank you Jasper!!

**Comment by kevinschlabach** [comment]
@jasoncypret @jasperfurniss @coleerikson - is there anything left for @stevelandwehr to do here?  or can we get approvals and merge this into playbook?  This is on our critical path for Runway's inline editing work...

**Review by coleerikson — APPROVED** [review]

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx
Comment (stevelandwehr) [reply playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx]: This is been resolved by using the `portalClassName` prop.

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**
